src/main/kotlin/stringcalculator/InputValidator.kt

```diff
@@ -0,0 +1,37 @@
+package stringcalculator
+
+class InputValidator {
+    fun validate(input: String?): List<String> {
+        val validString = validateInputIsNotEmpty(input)
+        val splitString = validString.split(" ")
+        return validateSplitStringIsCorrectlyStructured(splitString)
+    }
+
+    @Throws(IllegalArgumentException::class)
+    private fun validateInputIsNotEmpty(input: String?): String {
+        if (input.isNullOrBlank()) {
+            throw IllegalArgumentException("입력이 비어있습니다.")
+        }
+        return input
+    }
+
+    @Throws(IllegalArgumentException::class)
+    private fun validateSplitStringIsCorrectlyStructured(splitString: List<String>): List<String> {
+        when {
+            validateSize(splitString) -> throw IllegalArgumentException("잘못된 구성의 입력 입니다.")
+            validateStructure(splitString) -> throw IllegalArgumentException("잘못된 구성의 입력 입니다.")
+        }
+
+        return splitString
```

Comment:
```suggestion
        when {
            validateSize(splitString) || validateStructure(splitString) -> {
                throw IllegalArgumentException("잘못된 구성의 입력 입니다.")
            }
            else -> {
                return splitString
            }
        }
```
중복된 로직은 || 로 처리가 가능하고 else를 활용하면 when 에서 전부 처리할 수 있어요